### PR TITLE
feat: Create User Button in Header with Dropdown Menu

### LINKED ISSUE #61: Create User Button in Header with Dropdown Menu
## 📘 Description

Adding a user dropdown menu to the header enhances navigation by centralizing key actions like viewing the profile, managing listings, checking transaction history, and accessing settings. This improvement ensures a cleaner, more user-friendly interface, especially as more features are added.

## 🔢 Steps

1. Add the user button to the header.
2. Implement a dropdown menu that appears when the user button is clicked.
3. Include the following options with icons:
    - Profile
    - My Listings
    - Transaction History
    - Settings
4. Use **shadcn/ui** components for the dropdown.

## ✅ Acceptance Criteria

- The user button is added correctly to the header.
- The dropdown menu appears when the user button is clicked.
- The menu contains the listed options with corresponding icons.
- The design is responsive and aligns with the header design.

## 🌎 References

- [Lucide React Icons](https://lucide.dev/icons/)
- [shadcn/ui](https://ui.shadcn.com/docs)

## 📜 Additional Notes
- Blocked by: #54 

⚠ **Do not apply until the ODHack begins.**
⚠ **Read our guidelines before applying.**

### PR DESCRIPTION
**Pull request by johnnyzaet08**
# 📝  feat: Create User Button in Header with Dropdown Menu

## 🛠️ Issue
- Closes #61 

## 📖 Description
- The `dropdown-menu.tsx` component has been added, sourced from the official shadcn/ui documentation, to enable a user dropdown menu in the application’s header. To support this component, the `@radix-ui/react-dropdown-menu` dependency was installed. In addition, the `app/marketplace/page.tsx` file was updated to include the necessary imports for both the new dropdown menu and related icons, as well as the required JSX structure to display the user menu options. This integration now provides a cleaner, more intuitive way for users to access their profile, listings, transaction history, and settings directly from the header.

## ✅ Changes made
- Added the `dropdown-menu.tsx` component to the project.
- Installed and updated dependencies to ensure the dropdown functionality works as intended.
- Modified `app/marketplace/page.tsx` to incorporate the necessary imports and markup for the user dropdown menu.

## 🖼️ Media (screenshots/videos)
- The implementation of the User button is displayed
Screenshot 2024-12-13 132752
- The dropdown is displayed when the user button is clicked.
Screenshot 2024-12-13 132808
- The button is checked for responsive operation
Screenshot 2024-12-13 132727

## 📜 Additional Notes
- The implementation of another user within the header was taken into consideration, the structure of the user was respected and the new button was added within it.

### CODE REVIEW

File: frontend/app/marketplace/page.tsx
Comment (derianrddev) [inline frontend/app/marketplace/page.tsx]: Remove the `DropdownMenuLabel`

File: frontend/app/marketplace/page.tsx
Comment (derianrddev) [inline frontend/app/marketplace/page.tsx]: Remove the `DropdownMenuSeparator`

**Review by derianrddev — CHANGES_REQUESTED** [review]

**Comment by derianrddev** [comment]
@johnnyzaet08 Good job! Just some minor comments. Also, resolve the conflicts and add the user button to the new header, please.

**Review by derianrddev — APPROVED** [review]

**Comment by johnnyzaet08** [comment]
Hey @derianrddev, I need to wait for @danielcdz approval? Or can I do the merge right away? To solve the conflicts and make the merge.

**Comment by derianrddev** [comment]
Hey @johnnyzaet08 , you can go ahead and do the merge right now. No problem.

**Comment by danielcdz** [comment]
@johnnyzaet08 only maintainers can merge a PR! and this after receive at least 2 approvals on your PR

**Comment by johnnyzaet08** [comment]
Understood @danielcdz. 
Hey @derianrddev, I just did the update.  When I fixed the merge conflicts I got confused, then I realized and fixed it. Here is the test of the integration of the new button to the new header.

- Button integrated in the new header.
Screenshot 2024-12-15 222600

**Comment by derianrddev** [comment]
@johnnyzaet08 Thanks for the update, it looks good to me. We just need @danielcdz's approval to merge it into the project.

**Comment by danielcdz** [comment]
@johnnyzaet08 one last update to fix the conflicts and then we can merge!

**Review by danielcdz — APPROVED** [review]

**Pull request merged**
